### ISSUE TITLE
Fix for deadlock during build when latest CMS data fails accessibility tests

### ISSUE THREAD

**Issue by ncksllvn** [department-of-veterans-affairs/va.gov-team#802]
Merging this work is currently blocked by https://github.com/department-of-veterans-affairs/va.gov-team/issues/1065. The reason that ticket is blocking is because the work here increased our HTML validation and spotted two pages with issues as a result.

----

Currently, when there is invalid data from the CMS (such most commonly in the form of a broken link), the vets-website build falls back to a CMS cache, so that the wider engineering team can carry on unblocked. Without that, engineers wouldn't be able merge pull requests, and we'd have issues deploying.

However, the accessibility tests run during a later step of the build. This means that the latest CMS data could have compiled okay, but then fail later on in the build. At this point, it's too late to fall back to the CMS cache, which means everyone is blocked until the CMS team fixes the 508 issue on the erroneous page.

This is a weird issue. To fix this, we should explore possibly defining the accessibility tests as part of the Metalsmith build (vets-website code compilation), so that a page containing a 508 issue won't pass the compilation step, which would mean the vets-website build would correctly fall back to the CMS cache.

**Comment by ncksllvn** [comment]
Related ticket - https://github.com/department-of-veterans-affairs/va.gov-team/issues/294

By moving the accessibility tests into the build step, we can execute the tests on every environment.

**Comment by ncksllvn** [comment]
After some deep thinking, I am not sure if this is the right solution. No matter how fantastic our vets-website accessibility validation is, it's still validating content that lives far away from vets-website - in a Drupal database. 

As a testament to this, I cannot merge the associated work for this ticket until issue #1065 is resolved.

I may close this in favor of https://github.com/department-of-veterans-affairs/va.gov-team/issues/1466.

**Comment by ncksllvn** [comment]
I'm very sadly going to icebox this....I think the pull request I opened it is good, but it would make our tests stricter, and I think we need to rethink some things conceptually with how we validate data from the CMS. So, hesitantly, iceboxing....

**Comment by Karak888** [comment]
we are going to close this for now
